Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/UserMetadataServiceImpl.java

```diff
@@ -281,65 +280,94 @@ public void deleteUserMetadata(Integer usermetadataId, String username) throws U
-                    int count = 0;
-                    Set<UsermetadataLog> umlSet = u.getUsermetadataLog();
-                    if (umlSet.size() == logIds.size()) {  // May have duplicated UsermetadataLog umlSet.size()
-                        for (UsermetadataLog uml : umlSet) {
-                            if (logIds.contains(uml.getLog().getId())) {
-                                count += 1;
-                            }
-                        }
-                        if (count == logIds.size()) {
-                            result.add(u);
+                int count = 0;
+                Set<UsermetadataLog> umlSet = u.getUsermetadataLog();
+                if (umlSet.size() == logIds.size()) {  // May have duplicated UsermetadataLog umlSet.size()
+                    for (UsermetadataLog uml : umlSet) {
+                        if (logIds.contains(uml.getLog().getId())) {
+                            count += 1;
                         }
                     }
+                    if (count == logIds.size()) {
+                        result.add(u);
+                    }
                 }
             }
         }
 
+        return result;
+    }
+
+    @Override
+    public Set<Usermetadata> getUserMetadataByLog(Integer logId, UserMetadataTypeEnum userMetadataTypeEnum) {
+
+        Set<Usermetadata> umSet = new HashSet<>();
+        Set<UsermetadataLog> usermetadataLogSet =
+                new HashSet<>(usermetadataLogRepo.findByLog(logRepo.findUniqueByID(logId)));
+        for (UsermetadataLog ul : usermetadataLogSet) {
+            Usermetadata u = ul.getUsermetadata();
+            if (u.getUsermetadataType().getId().equals(userMetadataTypeEnum.getUserMetadataTypeId()) && u.getIsValid()) {
+                umSet.add(u);
+            }
+        }
+
+        return umSet;
+    }
+
+    public Set<Usermetadata> getUserMetadataByUserAndLog(String username, Integer logId,
+                                             UserMetadataTypeEnum userMetadataTypeEnum) throws UserNotFoundException {
+
+        return getUserMetadata(username, new ArrayList<>(logId), userMetadataTypeEnum);
+    }
+
+    @Override
+    public Set<Usermetadata> getUserMetadata(String username, List<Integer> logIds,
+                                             UserMetadataTypeEnum userMetadataTypeEnum) throws UserNotFoundException {
+
+        Set<Usermetadata> umSetLinkedToUser = getUserMetadataByUser(username, userMetadataTypeEnum);
+        Set<Usermetadata> umSetLinkedToLogs = getUserMetadataByLogs(logIds, userMetadataTypeEnum);
+
         // Find intersection of user metadata lists that get from 2 linked tables (log, group)
-        result.retainAll(usermetadataList1);
+        umSetLinkedToUser.retainAll(umSetLinkedToLogs);
 
-        return result.size() > 0 ? result : null;
+        return umSetLinkedToUser.size() > 0 ? umSetLinkedToUser : null;
     }
```

Comment:
Return an empty set would help avoid NPE at the calling side and make code less "if else" block.